reactor-core/src/main/java21/reactor/core/scheduler/LoomBoundedElasticScheduler.java

```diff
@@ -0,0 +1,1352 @@
+		 *
+		 * @see #release()
+		 * @see #dispose()
+		 */
+		void shutdown(boolean now) {
+			MARK_COUNT.set(this, now ? DISPOSED_NOW_STATE : DISPOSED_STATE);
+			int previousState = markShutdown(this);
+
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
+
+		/**
+		 * An alias for {@link #release()}.
+		 */
+		@Override
+		public void dispose() {
+			this.release();
+		}
+
+		Disposable schedule(Runnable task, @Nullable Composite disposables) {
+			if (isShutdown(this.state)) {
+				throw Exceptions.failWithRejected();
+			}
+
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			boolean isDirect = disposables == null;
+			SchedulerTask disposable =
+					new SchedulerTask(this, decoratedTask, -1, TimeUnit.NANOSECONDS, disposables);
+
+			if (!isDirect) {
+				if (!disposables.add(disposable)) {
+					throw Exceptions.failWithRejected();
+				}
+			}
+
+			this.tasksQueue.offer(disposable);
+
+			trySchedule();
+
+			return disposable;
+		}
+
+		Disposable schedule(Runnable task, long delay, TimeUnit unit, @Nullable Composite disposables) {
+			Objects.requireNonNull(unit, "TimeUnit should be non-null");
+
+			if (isShutdown(this.state)) {
+				throw Exceptions.failWithRejected();
+			}
+
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			boolean isDirect = disposables == null;
+			SchedulerTask disposable =
+					new SchedulerTask(this, decoratedTask, -1, unit, disposables);
+
+			if (!isDirect) {
+				if (!disposables.add(disposable)) {
+					throw Exceptions.failWithRejected();
+				}
+			}
+
+			if (delay <= 0) {
+				this.tasksQueue.offer(disposable);
+
+				trySchedule();
+			}
+			else {
+				disposable.schedule(delay, unit);
+			}
+
+			return disposable;
+		}
+
+		Disposable schedulePeriodically(Runnable task,
+				long initialDelay,
+				long period,
+				TimeUnit unit,
+				@Nullable Composite disposables) {
+			Objects.requireNonNull(unit, "TimeUnit should be non-null");
+
+			if (isShutdown(this.state)) {
+				throw Exceptions.failWithRejected();
+			}
+
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			boolean isDirect = disposables == null;
+
+			SchedulerTask disposable = new SchedulerTask(this,
+					decoratedTask,
+					period < 0 ? 0 : period,
+					unit,
+					disposables);
+
+			if (!isDirect) {
+				if (!disposables.add(disposable)) {
+					throw Exceptions.failWithRejected();
+				}
+			}
+
+			if (initialDelay <= 0) {
+				this.tasksQueue.offer(disposable);
+
+				trySchedule();
+			}
+			else {
+				disposable.schedule(initialDelay, unit);
+			}
+
+			return disposable;
+		}
+
+		void trySchedule() {
+			int previousState = addWork(this);
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
+
+		void tryForceSchedule() {
+			int previousState = forceAddWork(this);
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
```

Comment:
These two methods are used in different contexts. I think I understood when you explained it to me, but now I don't remember. I think it is worth documenting the inner behaviour properly for future maintainers.